OCP 4: syndesis-server exception: Caused by: java.net.UnknownHostException: openshift.default.svc: Name or service not known

## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>

## Description
On openshift 4, I see this error in the sydnesis-server logs:
`Caused by: java.net.UnknownHostException: openshift.default.svc: Name or service not known`

Using a terminal against the server pod:
```
sh-4.2$ ping openshift.default.svc
ping: openshift.default.svc: Name or service not known
sh-4.2$ ping kubernetes.default.svc
PING kubernetes.default.svc.cluster.local (172.30.0.1) 56(84) bytes of data.
From ip-10-131-0-1.ec2.internal (10.131.0.1) icmp_seq=1 Destination Host Unreachable
From ip-10-131-0-1.ec2.internal (10.131.0.1) icmp_seq=2 Destination Host Unreachable
From ip-10-131-0-1.ec2.internal (10.131.0.1) icmp_seq=3 Destination Host Unreachable```
```

--- a/app/server/openshift/src/main/java/io/syndesis/server/openshift/OpenShiftConfigurationProperties.java
+++ b/app/server/openshift/src/main/java/io/syndesis/server/openshift/OpenShiftConfigurationProperties.java
@@ -15,8 +15,6 @@
  */
 package io.syndesis.server.openshift;
 
-import java.nio.file.Files;
-import java.nio.file.Paths;
 import java.util.Map;
 
 import org.springframework.boot.context.properties.ConfigurationProperties;
@@ -29,11 +27,11 @@ import io.fabric8.openshift.client.OpenShiftConfigBuilder;
 @Validated
 public class OpenShiftConfigurationProperties {
 
-    public String serviceCaCertFile = "/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt";
-
     private boolean enabled;
 
-    private String openShiftHost = "https://kubernetes.default.svc";
+    private String masterUrlHost = "https://kubernetes.default.svc";
+
+    private OpenShiftConfig openShiftClientConfig = new OpenShiftConfigBuilder().withMasterUrl(masterUrlHost).build();
 
     private String builderImageStreamTag = "s2i-java:2.0";
 
@@ -72,20 +70,19 @@ public class OpenShiftConfigurationProperties {
     }
 
     public String getOpenShiftHost() {
-        return openShiftHost;
+        return masterUrlHost;
     }
 
     public void setOpenShiftHost(String openShiftHost) {
-        this.openShiftHost = openShiftHost;
+        this.masterUrlHost = openShiftHost;
+
+        if (!masterUrlHost.equals(openShiftHost)) {
+            openShiftClientConfig = new OpenShiftConfigBuilder().withMasterUrl(masterUrlHost).build();
+        }
     }
 
     public OpenShiftConfig getOpenShiftClientConfiguration() {
-        OpenShiftConfigBuilder builder = new OpenShiftConfigBuilder();
-        builder.withMasterUrl(openShiftHost);
-        if( Files.exists(Paths.get(serviceCaCertFile)) ) {
-            builder = builder.withCaCertFile(serviceCaCertFile);
-        }
-        return builder.build();
+        return openShiftClientConfig;
     }
 
     public String getBuilderImageStreamTag() {
@@ -175,13 +172,4 @@ public class OpenShiftConfigurationProperties {
     public void setBuildNodeSelector(Map<String, String> buildNodeSelector) {
         this.buildNodeSelector = buildNodeSelector;
     }
-
-    public String getServiceCaCertFile() {
-        return serviceCaCertFile;
-    }
-
-    public void setServiceCaCertFile(String serviceCaCertFile) {
-        this.serviceCaCertFile = serviceCaCertFile;
-    }
-
 }

